src/Workspaces/VisualBasic/Portable/Utilities/SpeculationAnalyzer.vb

```diff
@@ -518,21 +518,47 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Utilities
             Return node.IsKind(SyntaxKind.ParenthesizedExpression)
         End Function
 
-        Protected Overrides Function ConversionsAreCompatible(originalModel As SemanticModel, originalExpression As ExpressionSyntax, newModel As SemanticModel, newExpression As ExpressionSyntax) As Boolean
-            If originalExpression Is Nothing OrElse originalModel Is Nothing OrElse newExpression Is Nothing OrElse newModel Is Nothing Then
-                Return False
-            End If
+        Private Shared Function IsUnderNode(node As SyntaxNode, root As SyntaxNode) As Boolean
+            While node IsNot Nothing
+                If node Is root Then
+                    Return True
+                End If
+                If node.IsStructuredTrivia Then
+                    node = DirectCast(node, StructuredTriviaSyntax).ParentTrivia.Token.Parent
+                Else
+                    node = node.Parent
+                End If
+            End While
+            Return False
+        End Function
 
+        Protected Overrides Function ConversionsAreCompatible(originalModel As SemanticModel, originalExpression As ExpressionSyntax, newModel As SemanticModel, newExpression As ExpressionSyntax) As Boolean
             Return ConversionsAreCompatible(originalModel.GetConversion(originalExpression), newModel.GetConversion(newExpression))
         End Function
 
         Protected Overrides Function ConversionsAreCompatible(originalExpression As ExpressionSyntax, originalTargetType As ITypeSymbol, newExpression As ExpressionSyntax, newTargetType As ITypeSymbol) As Boolean
-            If originalExpression Is Nothing OrElse originalTargetType Is Nothing OrElse newExpression Is Nothing OrElse newTargetType Is Nothing Then
-                Return False
-            End If
+            Dim originalConversion As Conversion?
+            Dim newConversion As Conversion?
 
-            Dim originalConversion = Me.OriginalSemanticModel.ClassifyConversion(originalExpression, originalTargetType)
-            Dim newConversion = Me.SpeculativeSemanticModel.ClassifyConversion(newExpression, newTargetType)
+            If HasNonNullType(Me.OriginalSemanticModel, originalExpression) AndAlso
```

Comment:
The main problem is that [Conversion](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Binder/Semantics/Conversions/Conversion.cs) has no common type between VB and C#.  So we can't call ClassifyConversion from an abstract method.  What type would ClassifyConversion return?